### ISSUE TITLE
Clarifications regarding README file and YOLO-step-by-step.ipynb

### ISSUE THREAD

**Issue by Kzernobog** [experiencor/basic-yolo-keras#197]
Hi,
I am trying to train a custom class and have a few doubts regarding the code.

README:
- What do you mean by the statement "It supports training YOLOv2 network with various backends such as MobileNet and InceptionV3". Aren't YOLO, MobileNet and InceptionV3 different architectures for classification/detection? how does MobileNet and Inception act as backend to YOLOv2?
- The backend pre-trained weights that you mention, what dataset has that been trained upon?
- What exactly does the gen_anchors.py file do? If I'm not mistaken, does it generate the dimensions for the 5 individual anchor-boxes prescribed by YOLO. If so, would that change from one dataset to another?
- Can the custom class be any other class apart from the ones the model was pre-trained upon.

YOLO-step-by-step.ipynb
- can we use this notebook to train a custom class by adding the necessary images in the train/val folder, necessary annotations to the train_ann/val_ann folders, and adding a class name to LABELS variable in the notebook? or is the notebook just a demo to train on COCO dataset
- I realise that the pre-trained weights have been downloaded from the main YOLO site. Is the training donw only on the 24th layer, or are all layers trained using the pre-trained weights as initialization?
- Why is the notebook using the YOLOv1 architecture instead of the YOLOv2 architecture?

These might be super noob questions, please forgive the lack of knowledge.

**Comment by experiencor** [comment]
Hi @Kzernobog,

README:
1. These are very basic things. Please read the relevant papers by googling "YOLO", "MobileNet", and "InceptionV3". You will find that they are just different network architectures but all do the same thing: extract a hierarchy of features from the input image. These features can be used for very different purposes like classification and object detection.

2. They were trained on COCO2014.

3. That's right. Anchors depend on dataset.

4. Yes. You can use the pretrained weights to train a model to detect almost anything.

Step-by-Step

1. Yes, it can be used to train anything. One thing to note is that the notebook has no instructions on warming up. You can figure it out based on the instruction in the readme.

2. All the layers but only for detection on COCO.

3. No. It's YOLOv2.
